src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java

```
@@ -0,0 +1,91 @@
+package ru.yandex.practicum.filmorate.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+import ru.yandex.practicum.filmorate.exception.FilmValidationException;
+import ru.yandex.practicum.filmorate.model.Film;
+import ru.yandex.practicum.filmorate.storage.film.FilmStorage;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class FilmService {
+
+    private static final Logger logger = LoggerFactory.getLogger(FilmService.class);
+    FilmStorage filmStorage;
+
+    @Autowired
+    public FilmService(FilmStorage filmStorage) {
+        this.filmStorage = filmStorage;
+    }
+
+    public void add(Film film) {
+        if (!isValid(film)) {
+            logger.info(film + " не прошёл валидацию");
+            throw new FilmValidationException();
+        }
+        filmStorage.add(film);
+    }
+
+    public void update(@Valid @RequestBody Film film) {
```

Если валидацию можно произвести в контроллере, то лучше это сделать там. Чем раньше тем лучше - Fail Fast. Если запрос не корректный, то нам незачем тянуть его дальше. Особенно учитывая что в реальных приложениях нам часто приходится разделять DTO (с ними работает контроллер) и доменные сущности (с ними работает сервис) и производить конвертацию одного в другое не проверив корректен ли запрос просто не эффективно.